src/search/symbol.c

```diff
@@ -416,7 +416,7 @@ static symbol_relocation_t *Symbol_AddRelocation(
         assert(target != NULL);
         name_alloc = malloc(strlen(target) + 1);
         if (name_alloc != NULL) {
-            strncpy(name_alloc, target, strlen(target) + 1);
+            strcpy(name_alloc, target);

```

Comment:
Hmm, this looks like it's removing a check. Presumably the reason for the change is that the `strlen` was computed previously and should not be redone. It's fairly easy to see though in this situation that the check will always be true, but I would have preferred keeping the check to avoid future mistakes.

Reply:
Yeah, I agree that usually it'd be a good idea to check that, but it caused a compiler warning in its current form, and here in this case the check is, as you said, completely useless anyways.